lib/net.js

```diff
@@ -115,7 +115,7 @@ const {
   },
   genericNodeError,
 } = require('internal/errors');
-const { isUint8Array } = require('internal/util/types');
+const { isStringObject, isUint8Array } = require('internal/util/types');
```

Comment:
Nit: we rarely concern ourselves with string objects like this. An alternative approach would be to coerce the `ip` to a string using...

```js
isIP(`${ip}`)
```

Reply:
Makes sense. Especially because on C++ land the check uses the `IsString()`.

Reply:
I believe we still need to check for string object though. We want a function that returns `new String('127.0.0.1')` to be valid but not one that returns `['127.0.0.1']`

Just coercing would make both to be the same 

```js
`${new String('127.0.0.1')}` === `${['127.0.0.1']}` // true
```

Reply:
> We want a function that returns `new String('127.0.0.1')` to be valid.

Not necessarily. Most Node.js APIs use `validateString` and don't support String objects.

Reply:
I got it. Coercion will work then, and I'm okay with that. If we take this, we need to update the document, too.

Since `lookup` function should return a string I think we should point out that any provided object will be coerced to a string (aka `.toString()`)